packages/bitcoin-collateral-provider/lib/BitcoinCollateralProvider.js

```diff
@@ -1,573 +1,548 @@
-        '88', // OP_EQUALVERIFY
-        '76', 'a9', // OP_DUP OP_HASH160
-        '14', borrowerPubKeyHash, // OP_PUSHDATA(20) {alicePubKeyHash}
-        '88', 'ac', // OP_EQUALVERIFY OP_CHECKSIG
-      '67', // OP_ELSE
-        '63', // OP_IF
-          loanExpirationPushDataOpcode, // OP_PUSHDATA({loanExpirationHexLength})
-          loanExpirationHexEncoded, // {loanExpirationHexEncoded}
-          'b1', // OP_CHECKLOCKTIMEVERIFY
-          '75', // OP_DROP
-          '82', // OP_SIZE
-          '01', // OP_PUSHDATA(1)
-          '20', // Hex 32
-          '88', // OP_EQUALVERIFY
-          'a8', // OP_SHA256
-          '20', secretHashA2, // OP_PUSHDATA(32) {secretHashA2}
-          '88', // OP_EQUALVERIFY
-          '82', // OP_SIZE
-          '01', // OP_PUSHDATA(1)
-          '20', // Hex 32
-          '88', // OP_EQUALVERIFY
-          'a8', // OP_SHA256
-          '20', secretHashB2, // OP_PUSHDATA(32) {secretHashB2}
-          '88', // OP_EQUALVERIFY
-          '52', // PUSH #2
-          borrowerPubKeyPushDataOpcode, borrowerPubKey, // OP_PUSHDATA({alicePubKeyLength}) {alicePubKey}
-          lenderPubKeyPushDataOpcode, lenderPubKey, // OP_PUSHDATA({bobPubKeyLength}) {bobPubKey}
-          '52', // PUSH #2
-          'ae', // CHECKMULTISIG
-        '67', // OP_ELSE
-          '63', // OP_IF
-            biddingExpirationPushDataOpcode, // OP_PUSHDATA({expirationHexLength})
-            biddingExpirationHexEncoded, // {expirationHexEncoded}
-            'b1', // OP_CHECKLOCKTIMEVERIFY
-            '75', // OP_DROP
-            '82', // OP_SIZE
-            '01', // OP_PUSHDATA(1)
-            '20', // Hex 32
-            '88', // OP_EQUALVERIFY
-            'a8', // OP_SHA256
-            '20', secretHashA1, // OP_PUSHDATA(32) {secretHashA1}
-            '88', // OP_EQUALVERIFY
-            '76', 'a9', // OP_DUP OP_HASH160
-            '14', lenderPubKeyHash, // OP_PUSHDATA(20) {bobPubKeyHash}
-            '88', 'ac', // OP_EQUALVERIFY OP_CHECKSIG
-          '67', // OP_ELSE
-            seizureExpirationPushDataOpcode, // OP_PUSHDATA({seizureExpirationHexLength})
-            seizureExpirationHexEncoded, // {seizureExpirationHexEncoded}
-            'b1', // OP_CHECKLOCKTIMEVERIFY
-            '75', // OP_DROP
-            '76', 'a9', // OP_DUP OP_HASH160
-            '14', borrowerPubKeyHash, // OP_PUSHDATA(20) {alicePubKeyHash}
-            '88', 'ac', // OP_EQUALVERIFY OP_CHECKSIG
-          '68', // OP_ENDIF
-        '68', // OP_ENDIF
-      '68' // OP_ENDIF
-    ].join('')
+
+    const seizablePubKeyHash = seizable ? lenderPubKeyHash : borrowerPubKeyHash
+
+    return bitcoin.script.compile([
+      OPS.OP_IF,
+        OPS.OP_SIZE,
+        bitcoin.script.number.encode(32),
+        OPS.OP_EQUAL,
+        OPS.OP_SWAP,
+        OPS.OP_SHA256,
+        Buffer.from(secretHashB1, 'hex'),
+        OPS.OP_EQUAL,
+        OPS.OP_ADD,
+        OPS.OP_2,
+        OPS.OP_EQUAL,
```

Comment:
Both the value from `OP_EQUAL` for the length of the secret, and checking that the `sha256` value of the secret equalled the secretHash are added together, and it checks if it equals 2. Essentially, if both the length, and the secret is correct, output `OP_TRUE`.